### PR TITLE
feat: add table cell mat tooltip options

### PR DESCRIPTION
**Pull request by lab900-frederico-teixeira**

### CODE REVIEW

File: lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts [hunk 1]
```diff
@@ -4,15 +4,21 @@ import { TableCell } from '../../models/table-cell.model';
 @Component({
   selector: 'lab900-table-cell-value',
   template: ` <ng-container *ngIf="cell && cellValue">
-    <span *ngIf="!cell.click" [matTooltipClass]="'lab900-table__mat-tooltip'" [matTooltip]="getMatTooltip()">
+    <span
+      *ngIf="!cell.click"
+      matTooltipClass="lab900-table__mat-tooltip"
+      [matTooltip]="getMatTooltip()"
+      [matTooltipPosition]="this.cell?.cellTooltipOptions ?? 'below'"
```
Comment (lab900-winand-vandenbergh) [inline lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts:11]: @lab900-frederico-teixeira I believe below is the default position for the tooltip so you shouldn't have to specify it here.

File: lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts [hunk 2]
```diff
@@ -4,15 +4,21 @@ import { TableCell } from '../../models/table-cell.model';
...
       {{ cellValue | translate }}
     </span>
     <a
       style="cursor: pointer"
       *ngIf="cell.click"
       (click)="cell.click(data, cell)"
-      [matTooltipClass]="'lab900-table__mat-tooltip'"
+      matTooltipClass="lab900-table__mat-tooltip"
       [matTooltip]="getMatTooltip()"
+      [matTooltipPosition]="this.cell?.cellTooltipOptions ?? 'below'"
```
Comment (lab900-winand-vandenbergh) [inline lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts:21]: Same here

File: lib/src/lib/table/models/table-cell.model.ts [hunk 3]
```diff
@@ -66,4 +66,8 @@ export interface TableCell<T = any> {
    * Enable a tooltip, displays the cell content in a tooltip
    */
   cellTooltip?: (data: T, cell: TableCell) => string;
+  /**
+   * Options for the cellTooltip if available
+   */
+  cellTooltipOptions?: { tooltipPosition: string };
```
Comment (lab900-winand-vandenbergh) [inline lib/src/lib/table/models/table-cell.model.ts:72]: The tool tip positions are a fixed list of positions so maybe type it as an enum so no incorrect string can be passed?

**Review by lab900-winand-vandenbergh — CHANGES_REQUESTED** [review]

File: lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts (hunk 1 above)
Comment (lab900-frederico-teixeira) [reply lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts:11]: how should i do this 🙈 , passing null don't work

Comment (lab900-winand-vandenbergh) [reply lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts:11]: What about undefined? If that also doesn't work, i guess you'll have to default the cellTooltipOptions somewhere as 'below'. Just don't put it twice here in the template.

File: lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts [hunk 4]
```diff
@@ -56,4 +56,13 @@ export class Lab900TableCellValueComponent<T = any> implements OnChanges {
   public getMatTooltip(): string {
     return this.cell.cellTooltip ? this.cell.cellTooltip(this.data, this.cell) : '';
   }
+
+  public getMatTooltipOption(): MatTooltipOptions {
+    const matTooltipOption = this.cell.cellTooltipOptions ? this.cell.cellTooltipOptions : {};
+
+    return {
+      ...matTooltipOption,
+      tooltipPosition: matTooltipOption.tooltipPosition ?? 'below',
```
Comment (lab900-winand-vandenbergh) [inline lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts:65]: @lab900-frederico-teixeira Can you also get this from the TooltipPosition enum?

Comment (lab900-frederico-teixeira) [reply lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts:65]: I think, the typings are the only thing available, at least I didn't found then.

Comment (lab900-winand-vandenbergh) [reply lib/src/lib/table/components/table-cell-value/table-cell-value.component.ts:65]: Right, it's not really an enum, just typed as those strings. Fine with me then.

**Review by lab900-winand-vandenbergh — APPROVED** [review]

**Pull request merged**